src/cloudHandler.js

```diff
@@ -0,0 +1,46 @@
+var cloudURL;
+var cloudPath;
+
+/**
+ * Default sync cloud handler responsible for 
+ */
+var handler = function (params, success, failure) {
+    if (!cloudPath) {
+        // Default server sync api route
+        cloudPath = '/mbaas/sync/';
+    }
+    var url = cloudURL + cloudPath + params.dataset_id;
+    var json = JSON.stringify(params.req);
+
+    var xhr = new XMLHttpRequest();
+    xhr.open("POST", url, true);
+    xhr.setRequestHeader('Content-type', 'application/json; charset=utf-8');
+    xhr.onreadystatechange = function () {
+        if (xhr.readyState === XMLHttpRequest.DONE) {
+            if ((xhr.status >= 200 && xhr.status < 300) || xhr.status === 304) {
+                var responseJson;
+                try {
+                    if (xhr.responseText) {
+                        responseJson = JSON.parse(xhr.responseText);
+                    }
+                } catch (e) {
+                    return failure(e);
+                }
+                success(responseJson);
+            } else {
+                failure(xhr.responseText);
+            }
+        }
+    };
+    xhr.send(json);
```

Comment:
this function doesn't send the some of the app info into in the request, like app ids, client device ids etc. see https://github.com/feedhenry/fh-js-sdk/blob/master/src/modules/fhparams.js#L9. Is that deliberate?

Those information is specific to FH cloud apps, so a bit more discussion required here to decide if the default handler need to include the default request handler. Personally I think it shouldn't, and the fh-sync-js client then need to provide APIs to allow override. 

Also if this is the case, then the `cloudPath` should be specified by the user as well. The default `/mbaas/sync/` is also FH only.